api/queries_pr.go

```diff
@@ -799,6 +801,38 @@ func PullRequestReopen(client *Client, repo ghrepo.Interface, pr *PullRequest) e
 	return err
 }
 
+func PullRequestMerge(client *Client, repo ghrepo.Interface, pr *PullRequest) error {
+	return merge(client, repo, pr, githubv4.PullRequestMergeMethodMerge)
+}
+
+func PullRequestRebase(client *Client, repo ghrepo.Interface, pr *PullRequest) error {
+	return merge(client, repo, pr, githubv4.PullRequestMergeMethodRebase)
+}
+
+func PullRequestSquash(client *Client, repo ghrepo.Interface, pr *PullRequest) error {
+	return merge(client, repo, pr, githubv4.PullRequestMergeMethodSquash)
+}
+
+func merge(client *Client, repo ghrepo.Interface, pr *PullRequest, mergeMethod githubv4.PullRequestMergeMethod) error {
+	var mutation struct {
+		ReopenPullRequest struct {
```

Comment:
nit: copy-pasta

Reply:
A little scary that everything worked fine with that error!

Reply:
Ah, well, you can name your struct fields anything you like, and the GraphQL library will either request a field/connection/mutation based on that field name or based on the `graphql:"..."` struct tag. Since this field has a tag `graphql:"mergePullRequest(input: $input)"`, the mutation that is requested is always `mergePullRequest` and technically your overall query is correct regardless of the struct field name you choose.